replacement_scan.go

```diff
@@ -0,0 +1,65 @@
+package duckdb
+
+/*
+   #include <stdlib.h>
+   #include <duckdb.h>
+   void replacement_scan_cb(duckdb_replacement_scan_info info, const char *table_name, void *data);
+   typedef const char cchar_t;
+*/
+import "C"
+import (
+	"sync"
+	"unsafe"
+)
+
+type ReplacementScanCallback func(tableName string) (string, []any, error)
+
+var replacementScanFnMxt sync.Mutex
+var replacementScanCallback ReplacementScanCallback
```

Comment:
Maybe it is sensible to move this into a struct instead of individual variables for extensibility? Then, there can be a global `var` of that struct?